### ISSUE TITLE
Improve the naming and module organizations for a smoother / more ergonomic and intuitive interface

### ISSUE THREAD

**Issue by steph-lui** [getditto/safer_ffi#48]

**Comment by rustonaut** [comment]
Does that include names like `safer_ffi::slice::slice_boxed`?

I was wondering if there is a special reason it is not named `BoxedSlice`.

**Comment by danielhenrymantilla** [comment]
@rustonaut yes, that would be part of the changes. It will be internally renamed to `BoxedSlice`, and then publicly exposed through the `repr_c::Box<[_]>` shorthand, which will be extended much like it has just been for closures:

  - https://github.com/getditto/safer_ffi/pull/134
